src/tests/tribol_comp_geom.cpp

```diff
@@ -292,36 +292,321 @@ TEST_F( CompGeomTest, poly_area_centroid_2 )
+                                   tribol::PENALTY,
+                                   tribol::BINNING_GRID,
+                                   tribol::ExecutionMode::Sequential );
+
+   tribol::setPenaltyOptions( 0, tribol::KINEMATIC, tribol::KINEMATIC_CONSTANT );
+   tribol::setContactAreaFrac( 0, 1.e-4 );
+
+   RealT dt = 1.;
+   int update_err = tribol::update( 1, 1., dt );
+
+   EXPECT_EQ( update_err, 0 );
+
+   tribol::CouplingSchemeManager& couplingSchemeManager = 
+         tribol::CouplingSchemeManager::getInstance();
+
+   tribol::CouplingScheme* couplingScheme = 
+      &couplingSchemeManager.at( 0 );
+
+   EXPECT_EQ( 0, couplingScheme->getNumActivePairs() );
+
+}
+
+TEST_F( CompGeomTest, coincident_vertices_full_overlap )
+{
+   constexpr int dim = 2;
+   constexpr int numVerts = 2;
+   RealT xy1[dim*numVerts];
+   RealT xy2[dim*numVerts];
+
+   // this geometry should NOT be in contact
+   xy1[0] = 1.0;
+   xy1[1] = 0.0;
+   xy1[2] = 0.0;
+   xy1[3] = 0.0;
+
+   xy2[0] = 1.e-12;
+   xy2[1] = -0.1;
+   xy2[2] = 0.999999;
+   xy2[3] = -0.1;
+
+   RealT x1[numVerts];
+   RealT y1[numVerts];
+   RealT x2[numVerts];
+   RealT y2[numVerts];
+
+   for (int i=0; i<numVerts; ++i)
+   {
+      x1[i] = xy1[i*dim];
+      y1[i] = xy1[i*dim+1];
+      x2[i] = xy2[i*dim];
+      y2[i] = xy2[i*dim+1];
+   }
+
+   tribol::IndexT conn1[2] = {0,1};
+   tribol::IndexT conn2[2] = {0,1};
+
+   tribol::registerMesh( 0, 1, 2, &conn1[0], (int)(tribol::LINEAR_EDGE), &x1[0], &y1[0], nullptr, tribol::MemorySpace::Host );
+   tribol::registerMesh( 1, 1, 2, &conn2[0], (int)(tribol::LINEAR_EDGE), &x2[0], &y2[0], nullptr, tribol::MemorySpace::Host );
+
+   RealT fx1[2] = {0., 0.};
+   RealT fy1[2] = {0., 0.};
+   RealT fx2[2] = {0., 0.};
+   RealT fy2[2] = {0., 0.};
+
+   tribol::registerNodalResponse( 0, &fx1[0], &fy1[0], nullptr );
+   tribol::registerNodalResponse( 1, &fx2[0], &fy2[0], nullptr );
+
+   tribol::setKinematicConstantPenalty( 0, 1. );
+   tribol::setKinematicConstantPenalty( 1, 1. );
+
+   tribol::registerCouplingScheme( 0, 0, 1,
+                                   tribol::SURFACE_TO_SURFACE,
+                                   tribol::NO_CASE,
+                                   tribol::COMMON_PLANE,
+                                   tribol::FRICTIONLESS,
+                                   tribol::PENALTY,
+                                   tribol::BINNING_GRID,
+                                   tribol::ExecutionMode::Sequential );
+
+   tribol::setPenaltyOptions( 0, tribol::KINEMATIC, tribol::KINEMATIC_CONSTANT );
+   tribol::setContactAreaFrac( 0, 1.e-4 );
+
+   RealT dt = 1.;
+   int update_err = tribol::update( 1, 1., dt );
+
+   EXPECT_EQ( update_err, 0 );
+
+   tribol::CouplingSchemeManager& couplingSchemeManager = 
+         tribol::CouplingSchemeManager::getInstance();
+
+   tribol::CouplingScheme* couplingScheme = 
+      &couplingSchemeManager.at( 0 );
+
+   EXPECT_EQ( 1, couplingScheme->getNumActivePairs() );
```

Comment:
Ahh, if these should be in contact, the comment above should be updated.

Reply:
Fixed.